addons/html_editor/static/tests/syntax_highlighting.test.js

```diff
@@ -0,0 +1,1051 @@
+        stepFunction: async (editor) => {
+            await click("textarea");
+            const textarea = queryOne("textarea");
+            textarea.setSelectionRange(22, 22); // "    some\n       co    []de\n    for you"
+            await press(["shift", "tab"]);
+            await compareHighlightedContent(
+                getContent(editor.editable),
+                [
+                    {
+                        highlightedValue: `    some\n   co    de\n    for you`,
+                        textareaRange: 18, // "    some\n   co    []de\n    for you"
+                    },
+                ],
+                "The content was outdented a first time.",
+                editor
+            );
+            await press(["shift", "tab"]);
+        },
+        contentAfterEdit: {
+            highlightedValue: valueAfter,
+            textareaRange: 15, // "    some\nco    []de\n    for you"
+        },
+        contentAfter: `<pre data-language-id="plaintext">${valueAfter.replaceAll(
+            "\n",
+            "<br>"
+        )}</pre>[]`,
+        config: configWithEmbeddings,
+    });
+});
+
+test("shift+tab in selection in code block outdents each selected line", async () => {
+    await testEditor({
+        compareFunction: compareHighlightedContent,
+        contentBefore: "<pre>    a<br>    b c<br>     d</pre>",
+        contentBeforeEdit: { highlightedValue: "    a\n    b c\n     d" },
+        stepFunction: async (editor) => {
+            await click("textarea");
+            const textarea = queryOne("textarea");
+            textarea.setSelectionRange(5, 19); // "a[\nb c\n ]d"
+            await press(["shift", "tab"]);
+            await compareHighlightedContent(
+                getContent(editor.editable),
+                {
+                    highlightedValue: "a\nb c\n d",
+                    textareaRange: [1, 7], // "a[\nb c\n ]d"
+                },
+                "The content was outdented a first time.",
+                editor
+            );
+            // Remove the last remaining leading space.
+            await press(["shift", "tab"]);
+        },
+        contentAfterEdit: {
+            highlightedValue: "a\nb c\nd",
+            textareaRange: [1, 6], // "a[\nb c\n]d"
+        },
+        config: configWithEmbeddings,
+        contentAfter: `<pre data-language-id="plaintext">a<br>b c<br>d</pre>[]`,
+    });
+});
+
+test.tags("focus required");
+test("can switch between code blocks without issues", async () => {
+    const { editor } = await setupEditor(`<p>ab</p><pre>de</pre><p>gh</p><pre>jk</pre>`, {
+        config: configWithEmbeddings,
+    });
+    await compareHighlightedContent(
+        getContent(editor.editable),
+        [
+            "<p>ab</p>",
+            {
+                highlightedValue: "de",
+            },
+            "<p>gh</p>",
+            { highlightedValue: "jk" },
+        ],
```

Comment:
Is this supposed to be like `[expectedChild1, expectedChild2, ...]` ? Why not do something like:
````js
unformat(```
    <p>ab</p>
    ${ highlightedPre({ value: "de" }) }
    <p>gh</p>
    ${ highlightedPre({ value: "jk" }) }
```)
````
Isn't that more readable? And wouldn't that allow for removing `compareHighlightedContent` because the magic would happen in `highlightedPre` to return the expected html string?